include/openssl/opensslv.h

```diff
@@ -42,6 +42,11 @@ extern "C" {
 # define OPENSSL_VERSION_NUMBER  0x10101000L
 # define OPENSSL_VERSION_TEXT    "OpenSSL 1.1.1-dev  xx XXX xxxx"
 
+#define OPENSSL_MAKE_VERSION(maj,min,fix,patch,status) (((maj&0xf) << 28)+((min&0xff)<<20)+((fix&0xff)<<12)+((patch&0xff)<<4)+status)
```

Comment:
May I suggest that the names be `gen` (for "generattion"), `maj`, `min`, `edit` and `patch`?  That would conform much better with the version numbering we currently use.  If you wish, it's perfectly safe *for now and the foreseeable future* to assume that `gen == 1` (so you can replace `((maj&0xf) << 28)` with `0x10000000`)

Reply:
I guess I'm confused, are you saying that in version 1.2.3g, that "gen=1", "maj=2", "min=3", and patch=g?  If so, then I'll do exactly what you suggest...